packages/core/src/utils/shadowComparer.ts

```diff
@@ -0,0 +1,29 @@
+// shadow equality
+export const shadowComparer = <T>(
+  prev: T,
+  next: T,
+): boolean => {
+  const isDifferentTypes = typeof prev !== typeof next;
+
+  if (isDifferentTypes) {
+    return false;
+  }
+
+  const notReferenceTypes = !(prev instanceof Object) || !(next instanceof Object);
+  const isFunctionTypes = typeof prev === 'function' && typeof next === 'function';
```

Comment:
Either use 'is...' notation for all variables, or better -  let's drop it)
```suggestion
  const functionTypes = typeof prev === 'function' && typeof next === 'function';
```